option.go

```diff
@@ -194,6 +194,27 @@ func WithSchema(schema any) Option {
 	return &schemaOption{Schema: schema}
 }
 
+// InitializerFunc is a function that initializes a message. It may be used to
+// dynamically construct messages. It is called on client and handler receives
+// to construct the message to be unmarshaled into.
+//
+// The message will be a non nil pointer to the type created by the client or
+// handler. Use the Schema field of the [Spec] to determine the type of the
+// message.
+type InitializerFunc func(spec Spec, message any) error
```

Comment:
Does this exported type carry its weight? There aren't any methods on it - why not inline it into the two options below?

Reply:
This is address in #640